### PR TITLE
1-1 feature parity with Kotlin

### PR DESCRIPTION
**Pull request by ori-shem-tov**
- Added msgpack test
- Added `rawSign` and `signAlgoTransaction`
- Added more prefixes to check we don't sign other Algorand related stuff
- Removed prefixes check after decoding

### CODE REVIEW

File: assets/arc-0052/package.json [hunk 1]
```diff
@@ -15,11 +15,13 @@
     "@types/libsodium-wrappers-sumo": "^0.7.7",
     "@types/node": "^20.7.1",
     "ts-jest": "^29.1.1",
+    "tweetnacl": "^1.0.3",
     "typescript": "^5.2.2"
   },
   "dependencies": {
     "ajv": "^8.12.0",
     "algo-msgpack-with-bigint": "^2.1.1",
+    "algosdk": "^2.7.0",
```
Comment (ehanoc) [inline assets/arc-0052/package.json]: Can we not import `algosdk` it has too many third-party dependencies on it's own.

File: assets/arc-0052/contextual.api.crypto.spec.ts [hunk 2]
```diff
@@ -340,12 +358,23 @@ describe("Contextual Derivation & Signing", () => {
...
                     const metadata: SignMetadata = { encoding: Encoding.MSGPACK, schema: {} }
                     expect(cryptoService.signData(KeyContext.Identity,0, 0, transaction,  metadata)).rejects.toThrowError(ERROR_TAGS_FOUND)
                 })
             })
+        })
 
+        describe("signing transactions", () => {
+            it("\(OK) Sign Transaction", async () => {
+                const key: Uint8Array = await cryptoService.keyGen(KeyContext.Address, 0, 0)
+                // this transaction wes successfully submitted to the network https://testnet.explorer.perawallet.app/tx/UJG3NVCSCW5A63KPV35BPAABLXMXTTEM2CVUKNS4EML3H3EYGMCQ/
```
Comment (ehanoc) [inline assets/arc-0052/contextual.api.crypto.spec.ts:379]: What tx?

File: assets/arc-0052/contextual.api.crypto.ts [hunk 3]
```diff
@@ -8,18 +8,12 @@ import {
...
-    crypto_sign_ed25519_sk_to_pk,
-    crypto_scalarmult_ed25519_base,
     crypto_generichash,
-    crypto_scalarmult_base
 } from 'libsodium-wrappers-sumo';
 import * as msgpack from "algo-msgpack-with-bigint"
 import Ajv from "ajv"
 import { deriveChildNodePrivate, fromSeed } from './bip32-ed25519';
-import { randomBytes } from 'crypto';
+import { Transaction, EncodedSignedTransaction } from 'algosdk';
```
Comment (ehanoc) [inline assets/arc-0052/contextual.api.crypto.ts]: Let's not use algosdk Types. They couple the implementations together and it brings a lot more than just types, specially third-party dependencies.

File: assets/arc-0052/package.json (hunk 1 above)
Comment (ori-shem-tov) [reply assets/arc-0052/package.json]: so make `signAlgoTransaction` accept an encoded transaction?
we'll need to change the Kotlin impl to match this behaviour as well

File: assets/arc-0052/contextual.api.crypto.ts [hunk 4]
```diff
@@ -150,29 +185,44 @@ export class ContextualCryptoApi {
...
-        // h = hash(R || pubKey || msg) mod q
-        let h = crypto_core_ed25519_scalar_reduce(crypto_hash_sha512(Buffer.concat([R, publicKey, data])));
+        const bip44Path: number[] = GetBIP44PathFromContext(context, account, keyIndex)
 
-        // \(5): S = (r + h * k) mod q
-        const S = crypto_core_ed25519_scalar_add(r, crypto_core_ed25519_scalar_mul(h, scalar))
+        const sig =  await this.rawSign(bip44Path, prefixEncodedTx)
 
-        return Buffer.concat([R, S]);
+        const stx: EncodedSignedTransaction = {
```
Comment (ehanoc) [inline assets/arc-0052/contextual.api.crypto.ts]: Let's not use EncodedSignedTransaction.

**Review by ehanoc — CHANGES_REQUESTED** [review]

File: assets/arc-0052/contextual.api.crypto.spec.ts (hunk 2 above)
Comment (ori-shem-tov) [reply assets/arc-0052/contextual.api.crypto.spec.ts:379]: the same transaction that we're signing in the test

**Review by ehanoc — APPROVED** [review]

**Pull request merged**